compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14116,6 +14116,128 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
       return false;
    }
 
+TR::Register *
+OMR::Z::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *value = node->getChild(0);
+   TR::Node *addr = node->getChild(1);
+   TR::Node *length = node->getChild(2);
+
+   auto valueReg = cg->evaluate(value);
+   auto addrReg = cg->evaluate(addr);
+
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+   // Zero result reg
+   generateRRInstruction(cg, TR::InstOpCode::getXORRegOpCode(), node, resultReg, resultReg);
+
+   if (length->getOpCode().isLoadConst())
+      {
+      // Manage the constant length case
+      int arrayLen = length->getIntegerNodeValue<int>();
+      for (int x = 0; x < arrayLen; ++x)
+         {
+         // Load the shift amount into tmpReg
+         TR::MemoryReference *sourceMR = generateS390MemoryReference(addrReg, x, cg, NULL);
+         generateRXYInstruction(cg, TR::InstOpCode::LGB, node, tmpReg, sourceMR);
+
+         // Create memory reference using tmpReg (shift amount), then shift valueReg by shift amount
+         TR::MemoryReference * tempMR = generateS390MemoryReference(tmpReg, 0, cg);
```

Comment:
Can we get a better name for this memory reference (similarly in the `else` path)? Perhaps `shiftAmountMR`? Also would it be possible to fix the pointer location inconsistencies throughout this change? I'm seen all three flavors of `type* name`, `type * name`, and `type *name`. Let's stick to one.

Reply:
Addressed in: 62c4af1268e0b9753c9ba29ce364b7ec782aca17